pandas/parser.pyx

```diff
@@ -1468,6 +1505,97 @@ cdef _string_box_decode(parser_t *parser, int col,
 
     return result, na_count
 
+@cython.boundscheck(False)
+cdef _categorical_convert(parser_t *parser, int col,
```

Comment:
I am not sure I would do it this way. I would first parse the actual dtype of the column (e.g. into integer, datetimes, whatever), then just convert directly to a categorical. Not sure why this needs a whole factorizing step which is very duplicative.